tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,967 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+
+import argparse
+import constants
+import re
+import shutil
+import sys
+import os
+import yaml
+
+from functools import reduce
+from enum import Enum
+from cuda_to_hip_mappings import CUDA_TO_HIP_MAPPINGS
+
+
+def openf(filename, mode):
+    if sys.version_info[0] == 3:
+        return open(filename, mode, errors='ignore')
+    else:
+        return open(filename, mode)
+
+# Color coding for printing
+
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+class disablefuncmode(Enum):
+    """ How to disable functions
+    0 - Remove the function entirely (includes the signature).
+    1 - Stub the function and return an empty object based off the type.
+    2 - Add !defined(__HIP_PLATFORM_HCC__) preprocessors around the function.
+        This macro is defined by HIP if the compiler used is hcc.
+    3 - Add !defined(__HIP_DEVICE_COMPILE__) preprocessors around the function.
+        This macro is defined by HIP if either hcc or nvcc are used in the device path.
+    4 - Stub the function and throw an exception at runtime.
+    5 - Stub the function and throw an assert(0).
+    6 - Stub the function and keep an empty body.
+    """
+    REMOVE = 0
+    STUB = 1
+    HCC_MACRO = 2
+    DEVICE_MACRO = 3
+    EXCEPTION = 4
+    ASSERT = 5
+    EMPTYBODY = 6
+
+
+def update_progress_bar(total, progress):
+    """
+    Displays and updates a console progress bar.
+    """
+    barLength, status = 20, ""
+    progress = float(progress) / float(total)
+    if progress >= 1.:
+        progress, status = 1, "\r\n"
+
+    # Number of blocks to display. Used to visualize progress.
+    block = int(round(barLength * progress))
+    text = "\r[{}] {:.0f}% {}".format(
+        "#" * block + "-" * (barLength - block), round(progress * 100, 0),
+        status)
+
+    # Send the progress to stdout.
+    sys.stderr.write(text)
+
+    # Send the buffered text to stdout!
+    sys.stderr.flush()
+
+
+def filename_ends_with_extension(filename, extensions):
+    """Helper method to see if filename ends with certain extension"""
+    return reduce(lambda result, ext: filename.endswith("." + ext) or result, extensions, False)
+
+
+def inside_included_directories(dirpath, rootpath, include_dirs):
+    """Helper method to see if filename within included directories"""
+    return reduce(lambda result, included_directory: re.match(r'(%s)\b' % os.path.join(rootpath, included_directory), dirpath) or result, include_dirs, None)
```

Comment:
Stop using '%' for interpolation; use 'format' method instead.

Additionally, if you want to put a literal string into a regex, you have to escape it correctly using `re.escape`